src/core/unit-serialize.c

```diff
@@ -405,6 +406,98 @@ int unit_deserialize_state_skip(FILE *f) {
         }
 }
 
+int unit_deserialize_state_discard(FILE *f, FDSet *fds) {
+        int r;
+
+        assert(f);
+        assert(fds);
+
+        for (;;) {
+                _cleanup_free_ char *line = NULL, *key = NULL;
+                const char *val;
+
+                r = read_stripped_line(f, LONG_LINE_MAX, &line);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to read serialization line: %m");
+                if (r == 0)
+                        return 0;
+
+                if (isempty(line))
+                        return 1;
```

Comment:
Seems to not necessary to return `1`. The caller only checks if it is negative or not. Please use `0`.